### PR TITLE
Mapping dnceng repo to github id

### PR DESCRIPTION
**Pull request by Cosifne**
After we moved to dnceng, the diff change list is broken.

For example: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS/pullrequest/327472

The reason is, dnceng build from a minor of Roslyn, instead of Roslyn in Github. And our old code is supposing always get result from Github.

After this change:
https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/327644

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 1]
```diff
@@ -552,9 +552,13 @@ private static async Task<string> GetComponentBuildDropLogAsync(Build newestBuil
 
         internal static async Task<(List<GitCommit> changes, string diffLink)> GetChangesBetweenBuildsAsync(Build fromBuild, Build tobuild, CancellationToken cancellationToken)
         {
-            if (tobuild.Repository.Type == "GitHub")
+            var type = tobuild.Repository.Type;
+            if (type == "GitHub" || Options.ComponentBuildProjectName == "internal")
```
Comment (Cosifne) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs:556]: Also @JoeRobich,
Is this enough to distinguish dnceng release?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs [hunk 2]
```diff
@@ -27,6 +28,14 @@ public static partial class RoslynInsertionTool
         public static readonly Guid MLInfraSwatUserId = new Guid("6c25b447-1d90-4840-8fde-d8b22cb8733e");
         public static readonly Guid VSLSnapUserId = new Guid("9f64bc2f-479b-429f-a665-fec80e130b1f");
 
+        /// <summary>
+        /// Dnceng maintain and build from a copy of dotnet/roslyn. When trying to get the build from dnceng, it will
+        /// only return the id of the copy, not the real dotnet/roslyn in GitHub.
+        /// Therefore, we need this to map back to Roslyn
+        /// </summary>
+        public static readonly ImmutableDictionary<string, string> dncengRepoNameToGitHubId = ImmutableDictionary<string, string>
```
Comment (Cosifne) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs:36]: This is assuming whenever friend repos also moved to dnceng, they also need to add a value here

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 1 above)
Comment (JoeRobich) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs:556]: Please review https://github.com/dotnet/roslyn-tools/pull/982 which adds commit log generation for AzDO repos.

**Comment by JoeRobich** [comment]
Thanks @Cosifne for opening this. It was a good idea to substitute the GH repo name instead of only relying on the AzDO repo history. I worked the concept into my PR (#982) as a command line variable.

**Pull request closed without merging**